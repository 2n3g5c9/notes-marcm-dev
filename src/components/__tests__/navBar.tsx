import React from 'react'
import renderer from 'react-test-renderer'

import { NavBar } from '@components'

jest.mock('gatsby-plugin-mdx', () => {
  return {
    // eslint-disable-next-line react/display-name
    MDXRenderer: ({ children }: { children: React.ReactNode }) => {
      return <div>{children}</div>
    },
  }
})

describe('NavBar', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<NavBar />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
