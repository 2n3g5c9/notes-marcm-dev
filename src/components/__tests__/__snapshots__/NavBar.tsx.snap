// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<nav
  className="bg-gray-800"
>
  <div
    className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8"
  >
    <div
      className="flex items-center justify-between h-16"
    >
      <div
        className="flex items-center"
      >
        <a
          aria-label="Home"
          className="flex-shrink-0"
          href="/"
        >
          <test-file-stub
            alt="Title"
            className="lg:block h-8 w-auto"
          />
        </a>
        <div
          className="hidden sm:block sm:ml-6"
        >
          <div
            className="flex space-x-4"
          >
            <a
              aria-label="Home"
              className="text-gray-300 hover:bg-gray-700 hover:text-white px-3 py-2 rounded-md text-sm font-medium"
              href="/"
            >
              Home
            </a>
            <a
              aria-label="Notes"
              className="text-gray-300 hover:bg-gray-700 hover:text-white px-3 py-2 rounded-md text-sm font-medium"
              href="#"
            >
              Notes
            </a>
            <a
              aria-label="About"
              className="text-gray-300 hover:bg-gray-700 hover:text-white px-3 py-2 rounded-md text-sm font-medium"
              href="/about"
            >
              About
            </a>
          </div>
        </div>
      </div>
      <div
        className="flex items-center"
      >
        <button
          aria-checked={false}
          className="bg-accent relative inline-flex flex-shrink-0 p-0 h-6 w-11 border-2 border-transparent rounded-full cursor-pointer transition-colors ease-in-out duration-200 focus:outline-none"
          id="headlessui-switch-1"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="switch"
          tabIndex={0}
        >
          <span
            className="sr-only"
          >
            Toggle Dark Mode
          </span>
          <span
            className="translate-x-0 relative inline-block h-5 w-5 rounded-full bg-white shadow transform ring-0 transition ease-in-out duration-200"
          >
            <span
              aria-hidden="true"
              className="opacity-100 ease-in duration-200 absolute inset-0 h-full w-full flex items-center justify-center transition-opacity"
            >
              <test-file-stub
                className="h-3 w-3 text-accent"
                fill="currentColor"
              />
            </span>
            <span
              aria-hidden="true"
              className="opacity-0 ease-out duration-100 absolute inset-0 h-full w-full flex items-center justify-center transition-opacity"
            >
              <test-file-stub
                className="h-3 w-3 text-accent"
                fill="currentColor"
              />
            </span>
          </span>
        </button>
        <div
          className="text-gray-400 ml-3 -mr-2 sm:hidden"
        >
          <div
            aria-label="Toggle menu"
            className="hamburger-react"
            onClick={[Function]}
            onKeyUp={[Function]}
            role="button"
            style={
              Object {
                "cursor": "pointer",
                "height": "48px",
                "outline": "none",
                "position": "relative",
                "transform": "none",
                "transition": "0.4s cubic-bezier(0, 0, 0, 1)",
                "userSelect": "none",
                "width": "48px",
              }
            }
            tabIndex={0}
          >
            <div
              style={
                Object {
                  "background": "currentColor",
                  "borderRadius": "9em",
                  "height": "2px",
                  "left": "14px",
                  "position": "absolute",
                  "top": "17px",
                  "transform": "none",
                  "transition": "0.4s cubic-bezier(0, 0, 0, 1)",
                  "width": "20px",
                }
              }
            />
            <div
              style={
                Object {
                  "background": "currentColor",
                  "borderRadius": "9em",
                  "height": "2px",
                  "left": "14px",
                  "position": "absolute",
                  "top": "23px",
                  "transform": "none",
                  "transition": "0.4s cubic-bezier(0, 0, 0, 1)",
                  "width": "20px",
                }
              }
            />
            <div
              style={
                Object {
                  "background": "currentColor",
                  "borderRadius": "9em",
                  "height": "2px",
                  "left": "14px",
                  "position": "absolute",
                  "top": "29px",
                  "transform": "none",
                  "transition": "0.4s cubic-bezier(0, 0, 0, 1)",
                  "width": "20px",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="hidden sm:hidden"
  >
    <div
      className="px-2 pt-2 pb-3 space-y-1"
    >
      <a
        aria-label="Home"
        className="text-gray-300 hover:bg-gray-700 hover:text-white block px-3 py-2 rounded-md text-base font-medium"
        href="/"
      >
        🏡
          
        Home
      </a>
      <a
        aria-label="Notes"
        className="text-gray-300 hover:bg-gray-700 hover:text-white block px-3 py-2 rounded-md text-base font-medium"
        href="#"
      >
        📝
          
        Notes
      </a>
      <a
        aria-label="About️"
        className="text-gray-300 hover:bg-gray-700 hover:text-white block px-3 py-2 rounded-md text-base font-medium"
        href="/about"
      >
        👨🏻‍💻
          
        About️
      </a>
    </div>
  </div>
</nav>
`;
